// @uiw/react-baidu-map-label/cjs/useLabel.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useLabel = useLabel;
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = require("react");
var _reactBaiduMapUtils = require("@uiw/react-baidu-map-utils");
var _reactBaiduMapMap = require("@uiw/react-baidu-map-map");
function useLabel() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var _useState = (0, _react.useState)(),
    _useState2 = (0, _slicedToArray2.default)(_useState, 2),
    label = _useState2[0],
    setLabel = _useState2[1];
  var offset = props.offset,
    style = props.style,
    _props$content = props.content,
    content = _props$content === void 0 ? '' : _props$content,
    position = props.position,
    enableMassClear = props.enableMassClear;
  var _usePortal = (0, _reactBaiduMapUtils.usePortal)(),
    container = _usePortal.container,
    Portal = _usePortal.Portal;
  // const { container } = useRenderDom({ children: props.children });
  var _useMapContext = (0, _reactBaiduMapMap.useMapContext)(),
    map = _useMapContext.map;
  (0, _react.useEffect)(function () {
    if (!BMap || !map) return;
    if (!label) {
      var opts = {
        offset: offset,
        position: position,
        enableMassClear: enableMassClear
      };
      var instance = new BMap.Label(props.children ? container.innerHTML : content, opts);
      instance.setStyle((0, _objectSpread2.default)({}, style));
      map.addOverlay(instance);
      setLabel(instance);
    }
    return function () {
      if (map && label) {
        map.removeOverlay(label);
      }
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [map]);
  (0, _react.useEffect)(function () {
    if (label) {
      label.setContent(props.children ? container.innerHTML : content);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [props.children, container, label]);
  (0, _reactBaiduMapUtils.useVisiable)(label, props);
  (0, _reactBaiduMapUtils.useEventProperties)(label, props, ['Click', 'DblClick', 'MouseDo', 'MouseUp', 'MouseOout', 'MouseO', 'Remove', 'RightClick']);
  (0, _reactBaiduMapUtils.useProperties)(label, props, ['Style', 'Position', 'Offset', 'Title', 'ZIndex']);
  (0, _reactBaiduMapUtils.useEnableProperties)(label, props, ['MassClear']);
  return {
    label: label,
    setLabel: setLabel,
    Portal: Portal
  };
}
