// @uiw/react-baidu-map-drawing-manager/cjs/useDrawingManager.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useDrawingManager = useDrawingManager;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = require("react");
var _reactBaiduMapMap = require("@uiw/react-baidu-map-map");
var _reactBaiduMapUtils = require("@uiw/react-baidu-map-utils");
function useDrawingManager() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var isOpen = props.isOpen,
    drawingMode = props.drawingMode,
    enableDrawingTool = props.enableDrawingTool,
    enableCalculate = props.enableCalculate,
    drawingToolOptions = props.drawingToolOptions,
    markerOptions = props.markerOptions,
    circleOptions = props.circleOptions,
    polylineOptions = props.polylineOptions,
    polygonOptions = props.polygonOptions,
    rectangleOptions = props.rectangleOptions;
  var _useMapContext = (0, _reactBaiduMapMap.useMapContext)(),
    map = _useMapContext.map;
  var _useState = (0, _react.useState)(),
    _useState2 = (0, _slicedToArray2.default)(_useState, 2),
    drawingManager = _useState2[0],
    setDrawingManager = _useState2[1];
  var libSDK = window.BMapLib;
  var _useState3 = (0, _react.useState)(libSDK),
    _useState4 = (0, _slicedToArray2.default)(_useState3, 2),
    bMapLib = _useState4[0],
    setBMapLib = _useState4[1];
  var _useState5 = (0, _react.useState)(false || !!libSDK),
    _useState6 = (0, _slicedToArray2.default)(_useState5, 2),
    loadMapLib = _useState6[0],
    setLoadBMapLib = _useState6[1];
  var opts = {
    isOpen: isOpen,
    drawingMode: drawingMode,
    enableDrawingTool: enableDrawingTool,
    enableCalculate: enableCalculate,
    drawingToolOptions: drawingToolOptions,
    markerOptions: markerOptions,
    circleOptions: circleOptions,
    polylineOptions: polylineOptions,
    polygonOptions: polygonOptions,
    rectangleOptions: rectangleOptions
  };
  (0, _react.useMemo)(function () {
    // 如果第一次加载，会执行下面的
    if (map && bMapLib && !drawingManager) {
      if (bMapLib.DrawingManager) {
        var instance = new BMapLib.DrawingManager(map, opts);
        setDrawingManager(instance);
      }
    }

    // 如果 bMapLib 已经加载过，会执行下面的
    if (map && bMapLib && !bMapLib.DrawingManager) {
      (0, _reactBaiduMapUtils.requireCss)('//api.map.baidu.com/library/DrawingManager/1.4/src/DrawingManager_min.css').then(function () {});
      (0, _reactBaiduMapUtils.requireScript)('//api.map.baidu.com/library/DrawingManager/1.4/src/DrawingManager_min.js').then(function () {
        if (window.BMapLib) {
          var newMapLib = Object.assign(window.BMapLib, bMapLib);
          setBMapLib(newMapLib);
          var _instance = new BMapLib.DrawingManager(map, opts);
          setDrawingManager(_instance);
        }
      }).catch(function () {});
    }

    // 如果第一次加载，会执行下面的
    if (!bMapLib && !loadMapLib) {
      setLoadBMapLib(true);
      (0, _reactBaiduMapUtils.requireCss)('//api.map.baidu.com/library/DrawingManager/1.4/src/DrawingManager_min.css').then(function () {});
      (0, _reactBaiduMapUtils.requireScript)('//api.map.baidu.com/library/DrawingManager/1.4/src/DrawingManager_min.js').then(function () {
        if (window.BMapLib) {
          setBMapLib(window.BMapLib);
        }
      }).catch(function () {});
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [map, loadMapLib, bMapLib, drawingManager]);
  return {
    drawingManager: drawingManager,
    setDrawingManager: setDrawingManager,
    BMapLib: bMapLib
  };
}
