// @uiw/react-baidu-map-circle/cjs/useCircle.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useCircle = useCircle;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = require("react");
var _reactBaiduMapMap = require("@uiw/react-baidu-map-map");
var _reactBaiduMapUtils = require("@uiw/react-baidu-map-utils");
function useCircle() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var center = props.center,
    radius = props.radius,
    strokeColor = props.strokeColor,
    fillColor = props.fillColor,
    strokeWeight = props.strokeWeight,
    strokeOpacity = props.strokeOpacity,
    fillOpacity = props.fillOpacity,
    strokeStyle = props.strokeStyle,
    enableMassClear = props.enableMassClear,
    enableEditing = props.enableEditing,
    enableClicking = props.enableClicking;
  var _useMapContext = (0, _reactBaiduMapMap.useMapContext)(),
    map = _useMapContext.map;
  var _useState = (0, _react.useState)(),
    _useState2 = (0, _slicedToArray2.default)(_useState, 2),
    circle = _useState2[0],
    setCircle = _useState2[1];
  (0, _react.useMemo)(function () {
    if (map && !circle) {
      var point = center;
      if (center && center.lng && center.lat) {
        point = new BMap.Point(center.lng, center.lat);
      }
      var opts = {
        strokeColor: strokeColor,
        fillColor: fillColor,
        strokeWeight: strokeWeight,
        strokeOpacity: strokeOpacity,
        fillOpacity: fillOpacity,
        strokeStyle: strokeStyle,
        enableMassClear: enableMassClear,
        enableEditing: enableEditing,
        enableClicking: enableClicking
      };
      var instance = new BMap.Circle(point, radius, opts);
      map.addOverlay(instance);
      setCircle(instance);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [map, circle]);
  (0, _reactBaiduMapUtils.useVisiable)(circle, props);
  (0, _reactBaiduMapUtils.useEventProperties)(circle, props, ['Click', 'DblClick', 'MouseDown', 'MouseUp', 'MouseOut', 'MouseOver', 'Remove', 'LineUpdate']);
  (0, _reactBaiduMapUtils.useEnableProperties)(circle, props, ['Editing', 'MassClear']);
  (0, _reactBaiduMapUtils.useProperties)(circle, props, ['Center', 'Radius', 'StrokeColor', 'FillColor', 'StrokeOpacity', 'FillOpacity', 'StrokeWeight', 'StrokeStyle']);
  return {
    circle: circle,
    setCircle: setCircle
  };
}
