// @uiw/react-baidu-map-type-control/cjs/useMapTypeControl.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useMapTypeControl = useMapTypeControl;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = require("react");
var _reactBaiduMapMap = require("@uiw/react-baidu-map-map");
var _reactBaiduMapUtils = require("@uiw/react-baidu-map-utils");
function useMapTypeControl() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var _useState = (0, _react.useState)(),
    _useState2 = (0, _slicedToArray2.default)(_useState, 2),
    mapTypeControl = _useState2[0],
    setMapTypeControl = _useState2[1];
  var anchor = props.anchor,
    offset = props.offset,
    type = props.type,
    mapTypes = props.mapTypes;
  var _useMapContext = (0, _reactBaiduMapMap.useMapContext)(),
    map = _useMapContext.map;
  (0, _react.useEffect)(function () {
    if (map && !mapTypeControl) {
      var instance = new BMap.MapTypeControl({
        anchor: anchor || BMAP_ANCHOR_TOP_LEFT,
        offset: offset,
        type: type,
        mapTypes: mapTypes
      });
      map.addControl(instance);
      setMapTypeControl(instance);
      return function () {
        map.removeControl(instance);
      };
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [map]);
  (0, _reactBaiduMapUtils.useVisiable)(mapTypeControl, props);
  (0, _reactBaiduMapUtils.useProperties)(mapTypeControl, props, ['Anchor', 'Offset', 'Unit']);
  return {
    mapTypeControl: mapTypeControl,
    setMapTypeControl: setMapTypeControl
  };
}
