// @uiw/react-baidu-map-map/cjs/context.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.reducer = exports.initialState = exports.Context = void 0;
exports.useMapContext = useMapContext;
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _react = _interopRequireWildcard(require("react"));
var initialState = {
  map: undefined,
  BMap: undefined,
  container: undefined
};
exports.initialState = initialState;
var reducer = function reducer(state, action) {
  return (0, _objectSpread2.default)((0, _objectSpread2.default)({}, state), action);
};
exports.reducer = reducer;
var Context = /*#__PURE__*/_react.default.createContext({
  state: initialState,
  dispatch: function dispatch() {
    return null;
  }
});
exports.Context = Context;
function useMapContext() {
  var _useContext = (0, _react.useContext)(Context),
    state = _useContext.state,
    dispatch = _useContext.dispatch;
  return (0, _objectSpread2.default)((0, _objectSpread2.default)({}, state), {}, {
    state: state,
    dispatch: dispatch
  });
}
