// @uiw/react-baidu-map-map/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  Provider: true
};
exports.default = exports.Provider = void 0;
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = _interopRequireWildcard(require("react"));
var _useMap2 = require("./useMap");
Object.keys(_useMap2).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _useMap2[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _useMap2[key];
    }
  });
});
var _context = require("./context");
Object.keys(_context).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _context[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _context[key];
    }
  });
});
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["className", "style", "children"];
var Provider = function Provider(props) {
  var _useReducer = (0, _react.useReducer)(_context.reducer, _context.initialState),
    _useReducer2 = (0, _slicedToArray2.default)(_useReducer, 2),
    state = _useReducer2[0],
    dispatch = _useReducer2[1];
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_context.Context.Provider, {
    value: {
      state: state,
      dispatch: dispatch
    },
    children: props.children
  });
};
exports.Provider = Provider;
var _default = /*#__PURE__*/_react.default.forwardRef(function (_ref, ref) {
  var className = _ref.className,
    style = _ref.style,
    children = _ref.children,
    props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var _useReducer3 = (0, _react.useReducer)(_context.reducer, _context.initialState),
    _useReducer4 = (0, _slicedToArray2.default)(_useReducer3, 2),
    state = _useReducer4[0],
    dispatch = _useReducer4[1];
  window.BMap = window.BMap || window.BMapGL;
  var elmRef = (0, _react.useRef)(null);
  var _useMap = (0, _useMap2.useMap)((0, _objectSpread2.default)({
      container: elmRef.current
    }, props)),
    setContainer = _useMap.setContainer,
    container = _useMap.container,
    setCenter = _useMap.setCenter,
    setAutoLocalCity = _useMap.setAutoLocalCity,
    map = _useMap.map;
  // eslint-disable-next-line react-hooks/exhaustive-deps
  (0, _react.useEffect)(function () {
    return setContainer(elmRef.current);
  }, [elmRef.current]);
  // eslint-disable-next-line react-hooks/exhaustive-deps
  (0, _react.useMemo)(function () {
    return props.center && setCenter(props.center);
  }, [props.center]);
  // eslint-disable-next-line react-hooks/exhaustive-deps
  (0, _react.useEffect)(function () {
    return setAutoLocalCity(props.autoLocalCity);
  }, [props.autoLocalCity]);
  // eslint-disable-next-line react-hooks/exhaustive-deps
  (0, _react.useImperativeHandle)(ref, function () {
    return (0, _objectSpread2.default)((0, _objectSpread2.default)({}, props), {}, {
      map: map,
      BMap: BMap,
      container: elmRef.current
    });
  }, [map]);
  var childs = _react.default.Children.toArray(children);
  (0, _react.useEffect)(function () {
    if (map) {
      dispatch({
        map: map,
        container: elmRef.current,
        BMap: BMap
      });
    }
  }, [map]);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_context.Context.Provider, {
    value: {
      state: state,
      dispatch: dispatch
    },
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      ref: elmRef,
      className: className,
      style: (0, _objectSpread2.default)({
        fontSize: 1,
        height: '100%'
      }, style)
    }), BMap && map && typeof children === 'function' && children({
      BMap: BMap,
      map: map,
      container: container
    }), BMap && map && childs.map(function (child, key) {
      if (! /*#__PURE__*/_react.default.isValidElement(child)) return null;
      if (child.type && typeof child.type === 'string') {
        return /*#__PURE__*/_react.default.cloneElement(child, {
          key: key
        });
      }
      return /*#__PURE__*/_react.default.cloneElement(child, (0, _objectSpread2.default)((0, _objectSpread2.default)({}, child.props), {}, {
        BMap: BMap,
        map: map,
        container: container,
        key: key
      }));
    })]
  });
});
exports.default = _default;
