// @uiw/react-baidu-map/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  APILoader: true,
  Map: true,
  Circle: true,
  CurveLine: true,
  CustomOverlay: true,
  Marker: true,
  Label: true,
  TileLayer: true,
  InfoWindow: true,
  PointCollection: true,
  DrawingManager: true,
  Polyline: true,
  Polygon: true,
  withMap: true,
  RequireScript: true,
  Control: true,
  CopyrightControl: true,
  NavigationControl: true,
  GeolocationControl: true,
  OverviewMapControl: true,
  ScaleControl: true,
  MapTypeControl: true,
  PanoramaControl: true,
  GroundOverlay: true
};
Object.defineProperty(exports, "APILoader", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapApiLoader.default;
  }
});
Object.defineProperty(exports, "Circle", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapCircle.default;
  }
});
Object.defineProperty(exports, "Control", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapControl.default;
  }
});
Object.defineProperty(exports, "CopyrightControl", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapCopyrightControl.default;
  }
});
Object.defineProperty(exports, "CurveLine", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapCurveLine.default;
  }
});
Object.defineProperty(exports, "CustomOverlay", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapCustomOverlay.default;
  }
});
Object.defineProperty(exports, "DrawingManager", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapDrawingManager.default;
  }
});
Object.defineProperty(exports, "GeolocationControl", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapGeolocationControl.default;
  }
});
Object.defineProperty(exports, "GroundOverlay", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapGroundOverlay.default;
  }
});
Object.defineProperty(exports, "InfoWindow", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapInfoWindow.default;
  }
});
Object.defineProperty(exports, "Label", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapLabel.default;
  }
});
Object.defineProperty(exports, "Map", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapMap.default;
  }
});
Object.defineProperty(exports, "MapTypeControl", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapTypeControl.default;
  }
});
Object.defineProperty(exports, "Marker", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapMarker.default;
  }
});
Object.defineProperty(exports, "NavigationControl", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapNavigationControl.default;
  }
});
Object.defineProperty(exports, "OverviewMapControl", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapOverviewMapControl.default;
  }
});
Object.defineProperty(exports, "PanoramaControl", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapPanoramaControl.default;
  }
});
Object.defineProperty(exports, "PointCollection", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapPointCollection.default;
  }
});
Object.defineProperty(exports, "Polygon", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapPolygon.default;
  }
});
Object.defineProperty(exports, "Polyline", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapPolyline.default;
  }
});
Object.defineProperty(exports, "RequireScript", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapRequireScript.default;
  }
});
Object.defineProperty(exports, "ScaleControl", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapScaleControl.default;
  }
});
Object.defineProperty(exports, "TileLayer", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapTileLayer.default;
  }
});
Object.defineProperty(exports, "withMap", {
  enumerable: true,
  get: function get() {
    return _reactBaiduMapWithMap.default;
  }
});
var _reactBaiduMapUtils = require("@uiw/react-baidu-map-utils");
Object.keys(_reactBaiduMapUtils).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapUtils[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapUtils[key];
    }
  });
});
var _reactBaiduMapApiLoader = _interopRequireWildcard(require("@uiw/react-baidu-map-api-loader"));
Object.keys(_reactBaiduMapApiLoader).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapApiLoader[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapApiLoader[key];
    }
  });
});
var _reactBaiduMapMap = _interopRequireWildcard(require("@uiw/react-baidu-map-map"));
Object.keys(_reactBaiduMapMap).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapMap[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapMap[key];
    }
  });
});
var _reactBaiduMapCircle = _interopRequireWildcard(require("@uiw/react-baidu-map-circle"));
Object.keys(_reactBaiduMapCircle).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapCircle[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapCircle[key];
    }
  });
});
var _reactBaiduMapCurveLine = _interopRequireWildcard(require("@uiw/react-baidu-map-curve-line"));
Object.keys(_reactBaiduMapCurveLine).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapCurveLine[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapCurveLine[key];
    }
  });
});
var _reactBaiduMapCustomOverlay = _interopRequireWildcard(require("@uiw/react-baidu-map-custom-overlay"));
Object.keys(_reactBaiduMapCustomOverlay).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapCustomOverlay[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapCustomOverlay[key];
    }
  });
});
var _reactBaiduMapMarker = _interopRequireWildcard(require("@uiw/react-baidu-map-marker"));
Object.keys(_reactBaiduMapMarker).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapMarker[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapMarker[key];
    }
  });
});
var _reactBaiduMapLabel = _interopRequireWildcard(require("@uiw/react-baidu-map-label"));
Object.keys(_reactBaiduMapLabel).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapLabel[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapLabel[key];
    }
  });
});
var _reactBaiduMapTileLayer = _interopRequireWildcard(require("@uiw/react-baidu-map-tile-layer"));
Object.keys(_reactBaiduMapTileLayer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapTileLayer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapTileLayer[key];
    }
  });
});
var _reactBaiduMapInfoWindow = _interopRequireWildcard(require("@uiw/react-baidu-map-info-window"));
Object.keys(_reactBaiduMapInfoWindow).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapInfoWindow[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapInfoWindow[key];
    }
  });
});
var _reactBaiduMapPointCollection = _interopRequireWildcard(require("@uiw/react-baidu-map-point-collection"));
Object.keys(_reactBaiduMapPointCollection).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapPointCollection[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapPointCollection[key];
    }
  });
});
var _reactBaiduMapDrawingManager = _interopRequireWildcard(require("@uiw/react-baidu-map-drawing-manager"));
Object.keys(_reactBaiduMapDrawingManager).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapDrawingManager[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapDrawingManager[key];
    }
  });
});
var _reactBaiduMapPolyline = _interopRequireWildcard(require("@uiw/react-baidu-map-polyline"));
Object.keys(_reactBaiduMapPolyline).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapPolyline[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapPolyline[key];
    }
  });
});
var _reactBaiduMapPolygon = _interopRequireWildcard(require("@uiw/react-baidu-map-polygon"));
Object.keys(_reactBaiduMapPolygon).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapPolygon[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapPolygon[key];
    }
  });
});
var _reactBaiduMapWithMap = _interopRequireWildcard(require("@uiw/react-baidu-map-with-map"));
Object.keys(_reactBaiduMapWithMap).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapWithMap[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapWithMap[key];
    }
  });
});
var _reactBaiduMapRequireScript = _interopRequireDefault(require("@uiw/react-baidu-map-require-script"));
var _reactBaiduMapControl = _interopRequireWildcard(require("@uiw/react-baidu-map-control"));
Object.keys(_reactBaiduMapControl).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapControl[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapControl[key];
    }
  });
});
var _reactBaiduMapCopyrightControl = _interopRequireWildcard(require("@uiw/react-baidu-map-copyright-control"));
Object.keys(_reactBaiduMapCopyrightControl).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapCopyrightControl[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapCopyrightControl[key];
    }
  });
});
var _reactBaiduMapNavigationControl = _interopRequireWildcard(require("@uiw/react-baidu-map-navigation-control"));
Object.keys(_reactBaiduMapNavigationControl).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapNavigationControl[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapNavigationControl[key];
    }
  });
});
var _reactBaiduMapGeolocationControl = _interopRequireWildcard(require("@uiw/react-baidu-map-geolocation-control"));
Object.keys(_reactBaiduMapGeolocationControl).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapGeolocationControl[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapGeolocationControl[key];
    }
  });
});
var _reactBaiduMapOverviewMapControl = _interopRequireWildcard(require("@uiw/react-baidu-map-overview-map-control"));
Object.keys(_reactBaiduMapOverviewMapControl).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapOverviewMapControl[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapOverviewMapControl[key];
    }
  });
});
var _reactBaiduMapScaleControl = _interopRequireWildcard(require("@uiw/react-baidu-map-scale-control"));
Object.keys(_reactBaiduMapScaleControl).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapScaleControl[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapScaleControl[key];
    }
  });
});
var _reactBaiduMapTypeControl = _interopRequireWildcard(require("@uiw/react-baidu-map-type-control"));
Object.keys(_reactBaiduMapTypeControl).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapTypeControl[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapTypeControl[key];
    }
  });
});
var _reactBaiduMapPanoramaControl = _interopRequireWildcard(require("@uiw/react-baidu-map-panorama-control"));
Object.keys(_reactBaiduMapPanoramaControl).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapPanoramaControl[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapPanoramaControl[key];
    }
  });
});
var _reactBaiduMapGroundOverlay = _interopRequireWildcard(require("@uiw/react-baidu-map-ground-overlay"));
Object.keys(_reactBaiduMapGroundOverlay).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactBaiduMapGroundOverlay[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactBaiduMapGroundOverlay[key];
    }
  });
});
