// @uiw/react-baidu-map-geolocation-control/cjs/useGeolocationControl.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useGeolocationControl = useGeolocationControl;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = require("react");
var _reactBaiduMapMap = require("@uiw/react-baidu-map-map");
var _reactBaiduMapUtils = require("@uiw/react-baidu-map-utils");
function useGeolocationControl() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var _useState = (0, _react.useState)(),
    _useState2 = (0, _slicedToArray2.default)(_useState, 2),
    geolocationControl = _useState2[0],
    setGeolocationControl = _useState2[1];
  var anchor = props.anchor,
    offset = props.offset,
    showAddressBar = props.showAddressBar,
    enableAutoLocation = props.enableAutoLocation,
    locationIcon = props.locationIcon;
  var _useMapContext = (0, _reactBaiduMapMap.useMapContext)(),
    map = _useMapContext.map;
  (0, _react.useEffect)(function () {
    if (map && !geolocationControl) {
      var instance = new BMap.GeolocationControl({
        anchor: anchor || BMAP_ANCHOR_TOP_LEFT,
        offset: offset,
        showAddressBar: showAddressBar,
        enableAutoLocation: enableAutoLocation,
        locationIcon: locationIcon
      });
      map.addControl(instance);
      setGeolocationControl(instance);
      return function () {
        if (map && instance) {
          map.removeControl(instance);
        }
      };
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [map]);
  (0, _reactBaiduMapUtils.useVisiable)(geolocationControl, props);
  (0, _reactBaiduMapUtils.useEventProperties)(geolocationControl, props, ['LocationSuccess', 'LocationError'], 'CamelCase');
  (0, _reactBaiduMapUtils.useProperties)(geolocationControl, props, ['Anchor', 'Offset']);
  return {
    geolocationControl: geolocationControl,
    setGeolocationControl: setGeolocationControl
  };
}
